bin/cassandra.in.sh

```diff
@@ -122,13 +122,22 @@ jvmver=`echo "$java_ver_output" | grep '[openjdk|java] version' | awk -F'"' 'NR=
 JVM_VERSION=${jvmver%_*}
 short=$(echo "${jvmver}" | cut -c1-2)
 
-JAVA_VERSION=17
 if [ "$short" = "11" ]  ; then
      JAVA_VERSION=11
 elif [ "$JVM_VERSION" \< "17" ] ; then
     echo "Cassandra 5.0 requires Java 11 or Java 17."
     exit 1;
 fi
+# Allow execution if a supported Java version is used or CASSANDRA_USE_ALL_JDK argument is set
+java_versions_supported=11,17
+supported_version=$(echo "$java_versions_supported" | tr "," '\n' | grep -F -x "$short")
+
+if [ "x$CASSANDRA_USE_ALL_JDK" != "x" ] && [ "x$supported_version" = "x" ] ; then
```

Comment:
```suggestion
if [ -z "$CASSANDRA_USE_ALL_JDK" ] && [ -z "$supported_version" ] ; then
```
My testing shows that something like that should fix your startup problem. Tested with JDK11 and 17 - C* starts; JDK20 and 21 prints correctly the below message. 
So for the suggested warning, we need to add a print when the CASSANDRA_USE_ALL_JDK is set.